core/src/banking_stage/transaction_scheduler/prio_graph_scheduler.rs

```diff
@@ -197,20 +197,30 @@ impl PrioGraphScheduler {
     }
 
     /// Receive completed batches of transactions without blocking.
+    /// Returns (num_transactions, num_retryable_transactions) on success.
     pub fn receive_completed(
         &mut self,
         container: &mut TransactionStateContainer,
-    ) -> Result<(), SchedulerError> {
-        while self.try_receive_completed(container)? {}
-        Ok(())
+    ) -> Result<(usize, usize), SchedulerError> {
+        let mut total_num_transactions = 0;
+        let mut total_num_retryable = 0;
+        loop {
+            let (num_transactions, num_retryable) = self.try_receive_completed(container)?;
+            if num_transactions == 0 {
+                break;
+            }
+            total_num_transactions += num_transactions;
+            total_num_retryable += num_retryable;
+        }
+        Ok((total_num_transactions, total_num_retryable))
```

Comment:
Is it correct to say that `total_num_retryable` should significantly reduced, reflects how much `account-in-use` errors will be removed?

Reply:
Not quite. There are many other reasons we retry, aside from account-in-use. Some of which could potentially be increasing w/ scheduler changes.

For example, if we hit the end of our slot. The transactions are marked as retryable. Old code would hit one batch, and then stop iterating. With scheduler being a separate thread, it may have already queued more work than can actually be done since we don't have a reasonably accurate estimate of time per tx (maybe in the future)